File: LeetCode/2507. 使用质因数之和替换后可以取到的最小值.c
#include <stdio.h>

int smallestValue(int n)
{
    int pre = 0;
    while (n != pre)
    {
        pre = n;
        int sum = 0;
        for (size_t i = 2; i * i <= n; ++i)
        {
            while (n % i == 0)
            {
                sum += i;
                n /= i;
            }
        }
        sum += n > 1 ? n : 0;
        n = sum;
    }
    return n;
}

File: LeetCode/2507. 使用质因数之和替换后可以取到的最小值.js
function smallestValue(n) {
  let pre = 0
  while (n != pre) {
    pre = n
    let sum = 0
    for (let i = 2; i * i <= n; ++i) {
      while (n % i == 0) {
        sum += i
        n /= i
      }
    }
    sum += n > 1 ? n : 0
    n = sum
  }
  return n
}
console.log(smallestValue(15))
